packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -0,0 +1,111 @@
+import {
+    CheckoutService,
+    Extension,
+    ExtensionCommandType,
+    ExtensionRegion,
+} from '@bigcommerce/checkout-sdk';
+
+export class ExtensionService {
+    private _handlerRemovers: { [extensionId: string]: Array<() => void> } = {};
+
+    constructor(
+        private _checkoutService: CheckoutService,
+        private _setIsShowLoadingIndicator: (show: boolean) => void,
+    ) {}
+
+    async loadExtensions(): Promise<void> {
+        await this._checkoutService.loadExtensions();
+    }
+
+    async renderExtension(container: string, region: ExtensionRegion): Promise<void> {
+        const extension = this._getExtensionByRegion(region);
+
+        if (!extension) {
+            return;
+        }
+
+        await this._checkoutService.renderExtension(container, region);
+
+        if (!this._handlerRemovers[extension.id]) {
+            this._handlerRemovers[extension.id] = [];
+        }
+
+        this._handlerRemovers[extension.id].push(
+            this._checkoutService.handleExtensionCommand(
+                extension.id,
+                ExtensionCommandType.ReloadCheckout,
+                () => {
+                    void this._checkoutService.loadCheckout(
+                        this._checkoutService.getState().data.getCheckout()?.id,
+                    );
+                },
+            ),
+        );
+
+        this._handlerRemovers[extension.id].push(
```

Comment:
It's worth thinking about to how to extract the implementation of the handlers out, because I can imagine that this method can get quite long when we support more commands in the future.

Maybe each handler could be an exportable function on its own. And then we have a way of importing and registering them against each command type?

Reply:
I think this is a good starting point in my opinion, for extracting handlers and making things more generic we can probably do it as a fast follow once we deliver this desired functionality 🤔 We can even come back to this and re evaluate in more detail when we do come back to extensions as a more robust feature.

Reply:
Unless you think the proposed refactoring would take a lot of time, my recommendation is to do it now when things are still fresh in our minds so we have a scalable pattern to follow moving forward.

Reply:
I think we can do the refactoring as a fast follow in this case, so we can unblock the feature delivery and meet the agreements and make the improvements as a follow up.

Reply:
I believe this task is time consuming, but we can handle it in a separate ticket (https://bigcommercecloud.atlassian.net/browse/CHECKOUT-7635) following this PR. 
This way, we can share sample apps even before completing the entire project.

Reply:
> I believe this task is time consuming

Ok sure, I'm happy for you to do to tackle it separately if there's a contractual deadline we need to meet. 🙂 

Please reference the ticket number as a code comment here.

Reply:
I have put a TODO comment.
https://github.com/bigcommerce/checkout-js/blob/994a53343bd3dc4af0f2407e7cf33076088c88f2/packages/checkout-extension/src/ExtensionService.ts#L40